transitions/core.py

```diff
@@ -359,16 +356,17 @@ def __init__(self, model=None, states=None, initial=None, transitions=None,
         self._transition_queue = deque()
         self.models = []
 
-        if initial is None:
-            self.add_states('initial')
+        if model is None and add_self:  # TODO(pbovbel) reconsider API for next major release
+            model = self
+
```

Comment:
> Agreed re 0.5.x, in the meantime, can we keep add_self to maintain current behaviour/API but enable my use case?

sure, why not. with your additions, 0.5.0 is around the corner anyway :).

> I think the MachineError here makes sense as currently implemented (if no default 'initial' and no 'initial' in add_model)

yes, it does make sense if we allow Machine to not have a 'default' initial state. I am not sold on the idea we should let that happen.
But maybe we find a way to have `None` as a valid value for the initial parameter while keeping the current behaviour as default (e.g. `initial=False`?). If someone tries to intialize a Machine with a model and with `initial=None`an Exception is raised.